### PR TITLE
[OPIK-407] truncation bug fix

### PR DESCRIPTION
**Pull request by idoberko2**
## Details
The original truncation regex was too greedy, catching phrases that it should not catch. In addition, some of the image prefixes were inaccurate.

## Issues
OPIK-407

## Testing
Added a test to reproduce the reported bug.
In addition, performed local testing with various types of images to make sure everything is caught correctly.

### CODE REVIEW

File: apps/opik-backend/src/main/java/com/comet/opik/domain/ImageUtils.java [hunk 1]
```diff
@@ -4,24 +4,33 @@
...
-            + PREFIX_JPEG + IMAGE_CHARS + "|"
-            + PREFIX_PNG + IMAGE_CHARS + "|"
-            + PREFIX_GIF0 + IMAGE_CHARS + "|"
-            + PREFIX_GIF1 + IMAGE_CHARS + "|"
-            + PREFIX_BMP + IMAGE_CHARS + "|"
-            + PREFIX_TIFF0 + IMAGE_CHARS + "|"
-            + PREFIX_TIFF1 + IMAGE_CHARS + "|"
-            + PREFIX_WEBP + IMAGE_CHARS;
+    private static final String IMAGE_CHARS = "[A-Za-z0-9+/]+={0,2}";
+    private static final String IMAGE_TRUNCATION_REGEX =
```
Comment (ferc) [inline apps/opik-backend/src/main/java/com/comet/opik/domain/ImageUtils.java:16]: I think we can assume that the images are going to be always as a separate string field in the JSON, meaning that you could assume that it'll start and end with `"` adding in the middle all the image base64 combinations (`/"(data:image|iVBORw0KGgo|...).*"/`)

Comment (idoberko2) [reply apps/opik-backend/src/main/java/com/comet/opik/domain/ImageUtils.java:16]: Correct but then the `"` will be replaced as part of the "regex replace" function which will break the JSON structure

Comment (ferc) [reply apps/opik-backend/src/main/java/com/comet/opik/domain/ImageUtils.java:16]: Correct, then you can replace by `"[image]"` instead of just `[image]`

Comment (idoberko2) [reply apps/opik-backend/src/main/java/com/comet/opik/domain/ImageUtils.java:16]: Done in https://github.com/comet-ml/opik/pull/615/commits/1e4afb4456f01df1d2cb4a6cb8e565584d10acfa

**Review by thiagohora — APPROVED** [review]
LGTM

**Pull request merged**